lib/primitives/mtx.js

```diff
@@ -158,11 +164,19 @@ class MTX extends TX {
   addCoin(coin) {
     assert(coin instanceof Coin, 'Cannot add non-coin.');
 
-    const input = Input.fromCoin(coin);
+    // Avoid duplicate inputs
```

Comment:
`addCoin`, `addInput`, `addOutpoint` and `addTX` are utilities that are meant to be as simple as possible.
They don't do anything extra, pretty dumb methods. So complicating addCoin here I don't think is necessary, as you can always use pretty cheap check: `isSane` or `checkSanity` after you are done with everything.